File: python/courtlistener/api/attorneys.py
"""Attorneys API module for CourtListener SDK."""

from typing import Dict, Any, Optional
from ..utils.filters import build_filters
from ..utils.validators import validate_id
from ..models.base import BaseModel
from .base import BaseAPI


class Attorney(BaseModel):
    """Model for attorney data."""
    pass


class AttorneysAPI(BaseAPI):
    """API client for attorneys functionality."""
    
    def __init__(self, client):
        self.client = client
    
    def _get_endpoint(self) -> str:
        """Get the API endpoint for this module."""
        return "attorneys/"
    
    def _get_model_class(self):
        """Get the model class associated with this API."""
        return Attorney
    
    def list_attorneys(self, page: int = 1, q: str = None, **filters) -> Dict[str, Any]:
        """List attorneys with optional filtering."""
        params = filters.copy() if filters else {}
        params['page'] = page
        if q:
            params['q'] = q
        return self.client.get('attorneys/', params=params)
    
    def get_attorney(self, attorney_id: int) -> Attorney:
        """Get a specific attorney by ID."""
        validate_id(attorney_id)
        data = self.client.get(f'attorneys/{attorney_id}/')
        return Attorney(data)

    def search_attorneys(self, q: str, page: int = 1, **filters) -> Dict[str, Any]:
        params = filters.copy() if filters else {}
        params['q'] = q
        params['page'] = page
        return self.client.get('attorneys/', params=params) 

File: php/src/CourtListener/Api/Attorneys.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Attorneys API endpoint
 */
class Attorneys extends BaseApi
{
    protected string $endpoint = 'attorneys/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List attorneys with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listAttorneys(array $params = []): array
    {
        return $this->list($params);
    }

    /**
     * Get a specific attorneys by ID
     *
     * @param int|string $id Attorneys ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getAttorneys($id, array $params = []): array
    {
        return $this->get($id, $params);
    }

    /**
     * Search attorneys
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchAttorneys(array $params = []): array
    {
        return $this->search($params);
    }
}
